Docs/nightly_db_backup/db_backup.py

```diff
@@ -0,0 +1,117 @@
+			if file[-26:-20] < expiration_stamp:
+				os.remove(file)
+
+# MAIN -------------------------------------------------------------------------
+# ROOT required ----------------------------------------------------------------
+if os.getuid() != 0:
+	print ('Root required.  Please contact your sysadmin for assistance.', file=sys.stderr)
+	sys.exit(1)
+
+# DETERMINE CURRENT DATE -------------------------------------------------------
+# (today.year % 100) determines the two digit year.  e.g. '2017' -> '17'
+today       = datetime.date.today()
+year        = str(today.year % 100)
+today_stamp = '{:0>2}{:0>2}{:0>2}'.format(year, today.month, today.day)
+
+# DETERMINE CUURENT SEMESTER (based on current date) ---------------------------
+# if month <= 5: ... elif month >=8: ... else: ...
+semester = 's' + year if today.month <= 5 else ('f' + year if today.month >= 8 else 'm' + year)
+
+# GET FOLDER LIST (determines active courses) ----------------------------------
+courses_dir = '/var/local/submitty/courses/' + semester
+
+# Force lowercase and filter out all entries starting with '.'
+regex = re.compile('^(?!\.)')
+folder_list = filter(regex.match, [x.lower() for x in os.listdir(courses_dir)])
+re.purge()
+
+# BUILD LISTS AND PATH ---------------------------------------------------------
+db_list     = []
+dump_list   = []
+course_list = []
+
+for course in folder_list:
+
+	# Build lists used as pg_dump arguments
+	db_list.append('submitty_{}_{}'.format(semester, course))
+	dump_list.append('{}_{}_{}.dbdump'.format(today_stamp, semester, course))
+	course_list.append(course)
+
+	# Make sure backup folder exists for each course
+	dump_path = '{}/{}'.format(DUMP_PATH, course)
+	try:
+		os.mkdir(dump_path, 0o700)
+		os.chown(dump_path, 0, 0)
+	except OSError:
+		if not os.path.isdir(dump_path):
+			raise
+
+# DUMP -------------------------------------------------------------------------
+for i in range(len(course_list)):
+	# e.g. "pg_dump postgresql://user:password@host/dbname > /var/local/submitty-dump/course/dump_file.dbdump"
+	process = 'pg_dump postgresql://{}:{}@{}/{} > {}/{}/{}'.format(DB_USER, DB_PASS, DB_HOST, db_list[i], DUMP_PATH, course_list[i], dump_list[i])
+	return_code = subprocess.call(process, shell=True)
+
+	# If pg_dump doesn't return 0, display pg_dump return code and related course being dumped.
+	if return_code != 0:
+		print ('{}: pg_dump exited with error {}.'.format(course_list[i], return_code), file=sys.stderr)
+
+# DETERMINE EXPIRATION DATE (to delete obsolete dump files) --------------------
+# (do this BEFORE recursion so it is not calculated recursively n times)
+expiration       = datetime.date.fromordinal(today.toordinal() - 7)
+expiration_stamp = '{:0>2}{:0>2}{:0>2}'.format(expiration.year % 100, expiration.month, expiration.day)
+
+# RECURSIVELY CULL OBSOLETE DUMPS ----------------------------------------------
+delete_obsolete_dumps(DUMP_PATH, expiration_stamp)
+
+sys.exit(0)
```

Comment:
This is incredibly superficial.  Take it.  Fix it.  Reject it.  Up to you.  I have more important things to do than "fix" something that is both readable and working.